.github/workflows/puppet.yml

```diff
@@ -0,0 +1,47 @@
+name: Unit tests
+
+on: pull_request
+
+jobs:
+  unit:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby:
+          - "2.4"
+    name: Puppet
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-${{ matrix.ruby }}-gems-unit-${{ hashFiles('**/Gemfile.lock') }}
```

Comment:
we don't have a `Gemfile.lock` in the repo, so this cache will never hit. See https://github.com/theforeman/foreman-installer-modulesync/pull/106#issuecomment-718022711 for a similar discussion.